src/main/java/io/tiledb/spark/TileDBDataSourceReader.java

```diff
@@ -446,62 +482,73 @@ public boolean enableBatchRead() {
       // value
     } else if (filter instanceof EqualNullSafe) {
       EqualNullSafe f = (EqualNullSafe) filter;
-      int dimIndex = this.tileDBReadSchema.getDimensionId(f.attribute()).get();
-      ranges.get(dimIndex).add(new Range(new Pair<>(f.value(), f.value())));
+      int columnIndex = this.tileDBReadSchema.getColumnId(f.attribute()).get();
+      ranges.get(columnIndex).add(new Range(new Pair<>(f.value(), f.value())));
     } else if (filter instanceof EqualTo) {
       EqualTo f = (EqualTo) filter;
-      int dimIndex = this.tileDBReadSchema.getDimensionId(f.attribute()).get();
-      ranges.get(dimIndex).add(new Range(new Pair<>(f.value(), f.value())));
+      int columnIndex = this.tileDBReadSchema.getColumnId(f.attribute()).get();
+      ranges.get(columnIndex).add(new Range(new Pair<>(f.value(), f.value())));
 
       // GreaterThan is ranges which are in the form of `dim > 1`
     } else if (filter instanceof GreaterThan) {
       GreaterThan f = (GreaterThan) filter;
-      int dimIndex = this.tileDBReadSchema.getDimensionId(f.attribute()).get();
+      Object second;
+      if (nonEmptyDomain.get(f.attribute()) != null)
+        second = nonEmptyDomain.get(f.attribute()).getSecond();
+      else second = getMaxValue(f.value().getClass());
```

Comment:
style nit: newline and indent `second`

Reply:
All formatting is done by ./gradlew format.

Reply:
Ok, we can blame the robot 😄